pkg/workceptor/remote_work.go

```diff
@@ -68,17 +75,34 @@ func (rw *remoteUnit) connectToRemote(ctx context.Context) (net.Conn, *bufio.Rea
 }
 
 // getConnection retries connectToRemote until connected or the context expires
-func (rw *remoteUnit) getConnection(ctx context.Context) (net.Conn, *bufio.Reader) {
+func (rw *remoteUnit) getConnection(mw *utils.JobContext) (net.Conn, *bufio.Reader) {
 	connectDelay := utils.NewIncrementalDuration(SuccessWorkSleep, MaxWorkSleep, 1.5)
 	for {
-		conn, reader, err := rw.connectToRemote(ctx)
+		conn, reader, err := rw.connectToRemote(mw)
 		if err == nil {
 			return conn, reader
 		}
-		logger.Debug("Connection to %s failed with error: %s",
-			rw.Status().ExtraData.(*remoteExtraData).RemoteNode, err)
+		status := rw.Status()
+		logger.Warning("Connection to %s failed with error: %s",
+			status.ExtraData.(*remoteExtraData).RemoteNode, err)
+		errStr := err.Error()
+		if strings.Contains(errStr, "CRYPTO_ERROR") {
+			detailMsg := "Incorrect tlsclient to remote service"
```

Comment:
I think it would be a good idea to use fmt.Sprintf here to insert errStr into detailMsg.  Wherever possible, we want to pass error information up rather than mask it, to make life easier for anyone troubleshooting this in the future.

Reply:
fixed